samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+    SDL_Renderer *renderer;
+    SDL_Texture *sprite;
+    int window_w, window_h;
+    int sprite_w, sprite_h;
+    SDL_Event event;
+
+    /* Enable standard application logging */
+    SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
+
+    if (SDL_VideoInit(NULL) < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize SDL video: %s\n",
+                SDL_GetError());
+        _exit(1);
+    }
+
+    if (TTF_Init() < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize TTF: %s\n",
+                TTF_GetError());
+        _exit(1);
+    }      
+    
+    driver = SDL_GetCurrentVideoDriver();
+
+    window = SDL_CreateWindow("Demo",
+        SDL_WINDOWPOS_UNDEFINED,
+        SDL_WINDOWPOS_UNDEFINED,
+        SCREEN_WIDTH, SCREEN_HEIGHT,
+        SDL_WINDOW_SHOWN);
+    if(window == NULL)
+    {
+        debugPrint( "Window could not be created! SDL_Error: %s\n", SDL_GetError() );
+        _exit(1);
+    }
+
+    /* Create the renderer */
+    renderer = SDL_CreateRenderer(window, -1, 0);
+    if (!renderer) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create renderer: %s\n", SDL_GetError());
+        _exit(5);
+    }
+
+    /* Clear the window, load the sprite and go! */
+    SDL_SetRenderDrawColor(renderer, 0xA0, 0xA0, 0xA0, 0xFF);
+    SDL_RenderClear(renderer);
+
+    sprite = LoadSprite(renderer);
+    if (!sprite) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't load sprite: %s\n", SDL_GetError());
+        quit(6);
+    }
+
+    /* Allocate memory for the sprite info */
+    SDL_GetWindowSize(window, &window_w, &window_h);
+    SDL_QueryTexture(sprite, NULL, NULL, &sprite_w, &sprite_h);
+    positions = (SDL_Rect *) SDL_malloc(NUM_SPRITES * sizeof(SDL_Rect));
+    velocities = (SDL_Rect *) SDL_malloc(NUM_SPRITES * sizeof(SDL_Rect));
+    if (!positions || !velocities) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Out of memory!\n");
+        quit(2);
+    }
+
+    for (i = 0; i < NUM_SPRITES; ++i) {
+        positions[i].x = rand()*(i+1) % (window_w - sprite_w);
+        positions[i].y = rand()*(i+1) % (window_h - sprite_h);
+        positions[i].w = sprite_w;
+        positions[i].h = sprite_h;
+        velocities[i].x = 0;
+        velocities[i].y = 0;
+        while (!velocities[i].x && !velocities[i].y) {
+            velocities[i].x = (rand()*(i+1) % (MAX_SPEED * 2 + 1)) - MAX_SPEED;
+            velocities[i].y = (rand()*(i+1) % (MAX_SPEED * 2 + 1)) - MAX_SPEED;
+        }
+    }
+
+    /* Main render loop */
+    done = 0;
+    while (!done) {
+        /* Check for events */
+        while (SDL_PollEvent(&event)) {
+            switch (event.type) {
+            case SDL_WINDOWEVENT:
+                switch (event.window.event) {
+                case SDL_WINDOWEVENT_EXPOSED:
+                    SDL_SetRenderDrawColor(renderer, 0xA0, 0xA0, 0xA0, 0xFF);
+                    SDL_RenderClear(renderer);
```

Comment:
This feels weird `MoveSprites` will clear on every frame anyway (with another color).